lib/util.js

```diff
@@ -276,14 +285,44 @@ function formatValue(ctx, value, recurseTimes) {
     }
   }
 
-  var base = '', array = false, braces = ['{', '}'];
+  var base = '', empty, braces, formatter;
 
-  // Make Array say that they are Array
   if (Array.isArray(value)) {
-    array = true;
     braces = ['[', ']'];
+    empty = value.length === 0;
+    formatter = formatArray;
+  } else if (value instanceof Set) {
+    braces = ['Set {', '}'];
+    // With `showHidden`, `length` will display as a hidden property for
+    // arrays. For consistency's sake, do the same for `size`, even though this
+    // property isn't selected by Object.getOwnPropertyNames().
+    if (ctx.showHidden)
+      keys.unshift('size');
+    empty = value.size === 0;
+    formatter = formatSet;
+  } else if (value instanceof Map) {
+    braces = ['Map {', '}'];
+    // ditto
+    if (ctx.showHidden)
+      keys.unshift('size');
+    empty = value.size === 0;
+    formatter = formatMap;
+  } else {
+    // Only create a mirror if the object superficially looks like a Promise
+    var promiseInternals = value instanceof Promise && inspectPromise(value);
+    if (promiseInternals) {
+      braces = ['Promise {', '}'];
+      empty = false;
+      formatter = formatPromise;
+    } else {
+      braces = ['{', '}'];
+      empty = true; // no other data than keys
```

Comment:
Femto-style nit: two spaces before the comment and please capitalize and punctuate it.  Ditto for other comments.